src/Nethermind/Nethermind.Synchronization/SnapSync/SnapProviderHelper.cs

```diff
@@ -304,26 +317,30 @@ private static void StitchBoundaries(IList<TrieNode> sortedBoundaryList, ITrieSt
                         bool isBoundaryProofNode = false;
                         for (int ci = 0; ci <= 15; ci++)
                         {
-                            if (!IsChildPersisted(node, ci, store))
+                            if (!IsChildPersisted(node, ci, store, boundaryProofNodes))
                             {
                                 isBoundaryProofNode = true;
+                                boundaryProofNodes.Add(node);
                                 break;
                             }
                         }
 
-                        node.IsBoundaryProofNode = isBoundaryProofNode;
+                        if (isBoundaryProofNode == false)
```

Comment:
Shouldnt this be the other way around? If its a proof node, *then* add it to the hash?